truffle/src/main/java/org/jruby/truffle/nodes/core/InterpolatedStringNode.java

```diff
@@ -74,13 +74,14 @@ private Object concat(VirtualFrame frame, Object[] strings) {
 
         Object builder = null;
 
+        // TODO (nirvdrum 11-Jan-16) Rewrite to avoid massively unbalanced trees.
         for (Object string : strings) {
             assert RubyGuards.isRubyString(string);
 
             if (builder == null) {
                 builder = dupNode.call(frame, string, "dup", null);
             } else {
-                builder = concatNode.call(frame, builder, "concat", null, string);
+                builder = appendNode.call(frame, builder, "append", null, string);
```

Comment:
A Rubinius method/primitive. It does much of the same without some of the extra checking that `concat` needs to perform.